src/Features/CSharp/Portable/Highlighting/KeywordHighlighters/LoopHighlighter.cs

```diff
@@ -13,18 +11,25 @@
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Highlighting;
 using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Collections;
 using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.CSharp.KeywordHighlighting.KeywordHighlighters;
 
 [ExportHighlighter(LanguageNames.CSharp), Shared]
-internal class LoopHighlighter : AbstractKeywordHighlighter
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class LoopHighlighter() : AbstractKeywordHighlighter(findInsideTrivia: false)
 {
-    [ImportingConstructor]
-    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public LoopHighlighter()
-    {
-    }
+    protected override bool ContainsHighlightableToken(ref TemporaryArray<SyntaxToken> tokens)
+        => tokens.Any(static t => t.Kind()
+            is SyntaxKind.DoKeyword
+            or SyntaxKind.ForKeyword
+            or SyntaxKind.ForEachKeyword
+            or SyntaxKind.WhileKeyword
+            or SyntaxKind.BreakKeyword
+            or SyntaxKind.ContinueKeyword
+            or SyntaxKind.SemicolonToken);
```

Comment:
>SyntaxKind.SemicolonToken [](http://example.com/codeflow?start=15&length=25)

Can you clarify what this means in practice? If a semicolon is at the caret position, this highlighter will always be asked?

Reply:
yes.  that's correct.  that's because the loop highlighter will highlight if you're next to things like `break;`  that's ok.  this is still a tiny subset of all tokens.

Reply:
the main goal here is to stop runnin all these highlighters unnecessarily on tokens they will *never* report results for.  that's just pointless.